From e09e5a40a86826af7aeeaa926a44f09334b2f708 Mon Sep 17 00:00:00 2001
From: Alberto Murillo <alberto.murillo.silva@intel.com>
Date: Fri, 11 Mar 2016 15:54:53 -0600
Subject: [PATCH] stateless

---
 oslo_config/cfg.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/oslo_config/cfg.py b/oslo_config/cfg.py
index fc4274f..b38e429 100644
--- a/oslo_config/cfg.py
+++ b/oslo_config/cfg.py
@@ -1460,8 +1460,7 @@ class _ConfigDirOpt(Opt):
         def __call__(self, parser, namespace, values, option_string=None):
             """Handle a --config-dir command line argument.
 
-            :raises: ConfigFileParseError, ConfigFileValueError,
-                     ConfigDirNotFoundError
+            :raises: ConfigFileParseError, ConfigFileValueError
             """
             namespace._config_dirs.append(values)
             setattr(namespace, self.dest, values)
@@ -1469,7 +1468,7 @@ class _ConfigDirOpt(Opt):
             values = os.path.expanduser(values)
 
             if not os.path.exists(values):
-                raise ConfigDirNotFoundError(values)
+                return
 
             config_dir_glob = os.path.join(values, '*.conf')
 
-- 
2.5.0

